dev/sdk_python.go

```diff
@@ -28,28 +25,37 @@ type PythonSDK struct {
 	Dagger *DaggerDev // +private
 }
 
+func (t PythonSDK) dev(opts ...dagger.PythonSDKDevOpts) *dagger.PythonSDKDev {
+	src := t.Dagger.Source.Directory(pythonSubdir)
+	return dag.PythonSDKDev(src, opts...)
+}
+
+func (t PythonSDK) directory(dist *dagger.Directory) *dagger.Directory {
+	return dag.Directory().WithDirectory(pythonSubdir, dist)
+}
+
 // Lint the Python SDK
 func (t PythonSDK) Lint(ctx context.Context) error {
 	eg, ctx := errgroup.WithContext(ctx)
 
-	base := t.pythonBase(pythonDefaultVersion, true)
-
 	eg.Go(func() error {
-		path := "docs/current_docs"
-		_, err := base.
+		// Preserve same file hierarchy for docs because of extend rules in .ruff.toml
+		_, err := t.dev().
 			WithDirectory(
-				fmt.Sprintf("/%s", path),
-				t.Dagger.Source.Directory(path),
-				dagger.ContainerWithDirectoryOpts{
-					Include: []string{
-						"**/*.py",
-						".ruff.toml",
-					},
-				},
+				dag.Directory().
+					WithDirectory(
+						"",
+						t.Dagger.Source,
+						dagger.DirectoryWithDirectoryOpts{
+							Include: []string{
+								"docs/current_docs/**/*.py",
+								"**/.ruff.toml",
+							},
+						},
+					),
```

Comment:
This logic should move to a `docsDir` argument to `PythonSDKDev()`

Reply:
See https://github.com/dagger/dagger/pull/7720/files#diff-515b7e1d3dbaa7d9fd20e331311eb12d4d8a6a05aeb23ab69869aa02935307a8R59-R68 for a rough comparison

Reply:
That's one of the "rough edges" in this iteration that I plan on fixing in a different way later.

The reason it's here is because these docs are outside of `/sdk/python` and the config in `/docs` (`.ruff.toml`) extends the SDK's with relative paths. That's because there's custom config for linting in `/sdk/python/ruff.toml` that I want to reuse outside of the SDK's dir.

From the POV of `/sdk/python/dev`, it's not privy to the larger file structure of the repo. Note that there's a `/sdk/python/docs` which is supported by the dev module, but those are about the reference documentation.

The way I want to solve this, hopeful with _context directories_, is to make the config more neutral and use it globally for any directory that you pass in, which could be `/docs` or `/core/integration/testdata`. Then this part will be much more simplified.